src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -146,7 +147,10 @@ public override void Initialize()
                     // are re-entrant such that this won't cause problems,
                     // but could prevent issues of conflicting reports
                     // from servers that are quick to respond.
-                    var clusterDescription = Description.WithType(Settings.ConnectionMode.ToClusterType());
+#pragma warning disable 618
+                    var clusterDescription = Description.WithType(Settings.GetClusterType());
```

Comment:
Now, it's calculated based on `ConnectionMode`, `DirectConnection` and `replicaSetName`